app/atomic_svc.py

```diff
@@ -0,0 +1,253 @@
+
+        return string, payloads
+
+    def _handle_multiline_commands(self, cmd):
+        return cmd.replace('\n', ';')
+
+    async def _prepare_cmd(self, entries, test, platform, cmd):
+        '''
+        Handle a command or a cleanup (both are formatted the same way), given in `cmd`.
+        Return the cmd formatted as needed and payloads we need to take into account.
+        '''
+        payloads = []
+        cmd, new_payloads = self._use_default_inputs(entries, test, platform, cmd)
+        payloads.extend(new_payloads)
+        cmd, new_payloads = self._catch_path_to_atomics_folder(cmd)
+        payloads.extend(new_payloads)
+        cmd = self._handle_multiline_commands(cmd)
+        return cmd, payloads
+
+    async def _prepare_executor(self, entries, test, platform):
+        '''
+        Prepare the command and cleanup, and return them with the needed payloads.
+        '''
+        payloads = []
+
+        command, payloads_command = await self._prepare_cmd(entries, test, platform, test['executor']['command'])
+        cleanup, payloads_cleanup = await self._prepare_cmd(entries, test, platform, test['executor'].get('cleanup_command', ''))
+        payloads.extend(payloads_command)
+        payloads.extend(payloads_cleanup)
+
+        # TODO handle ART local files in command
+        # eg. https://github.com/redcanaryco/atomic-red-team/blob/a956d4640f9186a7bd36d16a63f6d39433af5f1d/atomics/T1022/T1022.yaml
+        # eg. https://github.com/redcanaryco/atomic-red-team/blob/fa67d8f04188955ec2c3d1532d2212f26ef5e3e8/atomics/T1502/T1502.yaml
+        return (command, cleanup, payloads)
+
+    async def _save_ability(self, entries, test):
+        ability_id = hashlib.md5(json.dumps(test).encode()).hexdigest()
+        at_ingested = 0
+        at_total = 0
+
+        tactics_li = self.technique_to_tactics.get(entries['attack_technique'], ['redcanary-unknown'])
+        tactic = 'multiple' if len(tactics_li) > 1 else tactics_li[0]
+
+        data = dict(
+            id=ability_id,
+            name=test['name'],
+            description=test['description'],
+            tactic=tactic,
+            technique=dict(
+                attack_id=entries['attack_technique'],
+                name=entries['display_name']
+            ),
+            platforms=dict()
+        )
+        for p in test['supported_platforms']:
+            at_total += 1
+            if test['executor']['name'] == 'manual':
+                # this test is expected to be run manually by a human, no automation is provided
+                continue
+
+            command, cleanup, payloads = await self._prepare_executor(entries, test, p)
+
+            executor = EXECUTORS.get(test['executor']['name'], 'unknown')
+            platform = PLATFORMS.get(p, 'unknown')
+            data['platforms'][platform] = dict()
+            data['platforms'][platform][executor] = dict(command=command, payloads=payloads, cleanup=cleanup)
+
+            at_ingested += 1
```

Comment:
Can make this a `!= 'manual'` and save a line I'd imagine

Reply:
fixed in a97b8f9